main/src/ca/uwaterloo/flix/language/errors/EntryPointError.scala

```diff
@@ -222,4 +222,150 @@ object EntryPointError {
+    }
+  }
+  /**
+    * An error raised to indicate that the expected type of the effect set of the function argument of a default handler is wrong.
+    *
+    * @param loc the location of the defn.
+    * @param eff the symbol of the effect associated with this default handler
+    */
+  case class ExpectedEffectVarForDefaultHandler(loc: SourceLocation, eff: Symbol.EffSym) extends EntryPointError {
+    def summary: String = s"Wrong signature for $eff's default handler found.\nExpected an effect variable for the effect of the function argument."
+
+    def message(formatter: Formatter): String = {
+      import formatter.*
+      s""">> Wrong signature for $eff's default handler found. Expected an effect variable for the effect of the function argument.
+         |
+         |${code(loc, "default handler.")}
+         |
+         |""".stripMargin
+    }
+  }
+  /**
+    * An error raised to indicate that the expected type of the argument of a default handler or its return type is wrong.
+    *
+    * @param loc the location of the defn.
+    * @param eff the symbol of the effect associated with this default handler
+    */
+  case class ExpectedTypeVarArgForDefaultHandler(loc: SourceLocation, eff: Symbol.EffSym) extends EntryPointError {
+    def summary: String = s"Wrong signature for $eff's default handler found.\nExpected the same type variable for both the return type of the function argument and the handler."
+
+    def message(formatter: Formatter): String = {
+      import formatter.*
+      s""">> Wrong signature for $eff's default handler found. Expected the same type variable for both the return type of the function argument and the handler.
+         |
+         |${code(loc, "default handler.")}
+         |
+         |""".stripMargin
+    }
+  }
+  /**
+    * An error raised to indicate that a default handler produces non primitive effects.
+    *
+    * @param loc the location of the defn.
+    * @param eff the symbol of the effect associated with this default handler
+    * @param wrongEff the symbol of the non primitive effect produced
+    */
+  case class NonPrimitiveEffectForDefaultHandler(loc: SourceLocation, eff: Symbol.EffSym, wrongEff: Symbol.EffSym) extends EntryPointError {
+    def summary: String = s"Wrong signature for $eff's default handler found.\nNon Primitive effect $wrongEff generated."
+
+    def message(formatter: Formatter): String = {
+      import formatter.*
+      s""">> Wrong signature for $eff's default handler found. Non Primitive effect $wrongEff generated.
+         |
+         |${code(loc, "default handler.")}
+         |
+         |""".stripMargin
+    }
+  }
+  /**
+    * An error raised to indicate that there are multiple default handlers for the same effect.
+    *
+    * @param loc the location of the defn.
+    * @param eff the effect associated with this default handler
+    */
+  case class MultipleDefaultHandlers(loc: SourceLocation, eff: Type) extends EntryPointError {
```

Comment:
Can you look at some of the other duplicate errors? I think we should want the source location of both handlers.

Reply:
We do all pairs, I guess you get at most nˆ2?